python/paddle/v2/fluid/tests/book/test_recognize_digits.py

```diff
@@ -137,13 +138,49 @@ def main():
                 acc_val = numpy.array(acc_set).mean()
                 avg_loss_val = numpy.array(avg_loss_set).mean()
                 if float(acc_val) > 0.85:  # test acc > 85%
-                    exit(0)
+                    if save_dirname is not None:
+                        fluid.io.save_inference_model(save_dirname, ["img"],
+                                                      [prediction], exe)
+                    return
                 else:
                     print(
                         'PassID {0:1}, BatchID {1:04}, Test Loss {2:2.2}, Acc {3:2.2}'.
                         format(pass_id, batch_id + 1,
                                float(avg_loss_val), float(acc_val)))
 
 
+def infer(args, save_dirname=None):
+    if save_dirname is None:
+        return
+
+    place = fluid.CUDAPlace(0) if args.use_cuda else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+
+    # Use fluid.io.load_inference_model to obtain the inference program desc,
+    # the feed_target_names (the names of variables that will be feeded 
+    # data using feed operators), and the fetch_targets (variables that 
+    # we want to obtain data from using fetch operators).
+    [inference_program, feed_target_names,
+     fetch_targets] = fluid.io.load_inference_model(save_dirname, exe)
+
+    if args.nn_type == 'mlp':
+        tensor_img = numpy.random.rand(1, 28, 28).astype("float32")
+    else:
+        tensor_img = numpy.random.rand(1, 1, 28, 28).astype("float32")
```

Comment:
Why are the numpy array dimensions different for 'mlp' and 'conv_net' ?

Reply:
Because the input's dimension of `conv_op` should be 4-D or 5-D, and the input's dimension of `mul_op` will be flattened into 2-D. So, we can use `line 169` to simplify the code.